pkg/services/folder/folderimpl/folder.go

```diff
@@ -253,17 +254,53 @@ func (s *Service) Create(ctx context.Context, cmd *folder.CreateFolderCommand) (
 	return folder.FromDashboard(dash), nil
 }
 
-func (s *Service) UpdateFolder(ctx context.Context, user *user.SignedInUser, orgID int64, existingUid string, cmd *models.UpdateFolderCommand) error {
+func (s *Service) Update(ctx context.Context, user *user.SignedInUser, orgID int64, existingUid string, cmd *models.UpdateFolderCommand) (*models.Folder, error) {
+	foldr, err := s.legacyUpdate(ctx, user, orgID, existingUid, cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	if s.features.IsEnabled(featuremgmt.FlagNestedFolders) {
+		if cmd.Uid != "" {
+			if !util.IsValidShortUID(cmd.Uid) {
+				return nil, dashboards.ErrDashboardInvalidUid
```

Comment:
This can wait for a future PR, but we should make  folder service specific errors - we've got some already:
 https://github.com/grafana/grafana/blob/230942f008356c19ff119dd5692418fd346ed1a2/pkg/services/folder/model.go#L10

Reply:
I will merge and address this on the following PR